src/App/Entity/Proposal.php

```diff
@@ -208,22 +252,23 @@ public function addArticle(Article $article)
         return $this;
     }
 
-    public function removeArticle(Article $article)
-    {
-        $this->articles->removeElement($article);
-        $article->removeProposal($this);
-
-        return $this;
-    }
-
     public function getArticles()
     {
         return $this->articles;
     }
 
-    public function getNumberOfArticles()
+    public function getNumberOfPublishedArticles()
     {
-        return $this->articles->count();
+        $articles                  = $this->getArticles();
+        $numberOfPublishedArticles = 0;
+
+        foreach ($articles as $article) {
```

Comment:
La variable $articles est pas très utile ici

`foreach ($this->getArticles() as $article) {`

Reply:
done